overlord/snapstate/snapmgr_test.go

```diff
@@ -233,3 +235,24 @@ func (s *snapmgrTestSuite) TestSetInactive(c *C) {
 	c.Assert(s.fakeBackend.ops[0].name, Equals, "some-snap-to-inactivate")
 	c.Assert(s.fakeBackend.ops[0].active, Equals, false)
 }
+
+func (s *snapmgrTestSuite) TestSnapInfo(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	dirs.SetRootDir(c.MkDir())
+	defer dirs.SetRootDir("")
+
+	// Write a snap.yaml with fake name
+	dname := filepath.Join(dirs.SnapSnapsDir, "samba", "123", "meta")
+	err := os.MkdirAll(dname, 0775)
+	c.Assert(err, IsNil)
+	fname := filepath.Join(dname, "snap.yaml")
+	err = ioutil.WriteFile(fname, []byte("name: ---\n"), 0644)
+	c.Assert(err, IsNil)
+
+	// Ensure that name is overridden
+	snapInfo, err := snapstate.SnapInfo(s.state, "samba", "123")
+	c.Assert(err, IsNil)
+	c.Check(snapInfo.Name, Equals, "samba")
```

Comment:
This is the one field that is manually set. You can drop all the rest of the logic in SnapInfo besides that one line and the test would pass, right?